graphrag_sdk/models/azure_openai.py

```diff
@@ -227,3 +227,22 @@ def _get_generation_config(self, output_method: OutputMethod):
             config['response_format'] = { "type": "json_object" }
         
         return config
+    
+    def delete_last_message(self):
+        """
+        Deletes the last message exchange (user message and assistant response) from the chat history.
+        Preserves the system message if present.
+        
+        Note: Does nothing if the chat history is empty or contains only a system message.
+        """
+        # Keep at least the system message if present
+        min_length = 1 if self._model.system_instruction else 0
+        if len(self._history) - 2 >= min_length:
+            self._history = self._history[:-2]
```

Comment:
might be more efficient cheaper to just call: `self._history.pop()` twice removing the last two elements.